pkg/pathutils/pathutils.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright © 2021 Aspect Build Systems Inc
+Not licensed for re-use.
+*/
+
+package pathutils
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+)
+
+// https://github.com/bazelbuild/bazel/blob/8346ea4c/src/main/cpp/workspace_layout.cc#L37
+var workspaceFilenames = []string{"WORKSPACE", "WORKSPACE.bazel"}
+
+type RunFn func(cmd *cobra.Command, args []string) (exitErr error)
+
+func InvokeCmdInsideWorkspace(toRun RunFn) (wrapped RunFn) {
+	return func(cmd *cobra.Command, args []string) (exitErr error) {
+		finder := NewDefaultWorkspaceFinder()
+		cwd, err := os.Getwd()
+		if err != nil {
+			return fmt.Errorf("failed to run command %q inside workspace: %w", cmd.Use, err)
+		}
+		workspacePath, err := finder.Find(cwd)
+		if err != nil {
+			return fmt.Errorf("failed to run command %q inside workspace: %w", cmd.Use, err)
+		}
+		if workspacePath == "" {
+			err = fmt.Errorf("the current working directory %q is not a bazel workspace " +
+				"(below a directory having a WORKSPACE file)", cwd)
+			return fmt.Errorf("failed to run command %q inside workspace: %w", cmd.Use, err)
+		}
+		return toRun(cmd, args)
+	}
+}
+
+// WorkspaceFinder is the interface that wraps the simple Find method that performs the
+// finding of the WORKSPACE file in the user's Bazel project.
+type WorkspaceFinder struct {
+	osGetwd func() (string, error)
+	osStat  func(string) (fs.FileInfo, error)
+}
+
+// NewDefaultWorkspaceFinder instantiates a default internal implementation of the WorkspaceFinder
```

Comment:
It's not an internal implementation since the `WorkspaceFinder` is an exported struct (WorkspaceFinder vs workspaceFinder).